src/server/rdb_load.cc

```diff
@@ -1708,6 +1760,36 @@ auto RdbLoaderBase::LoadLen(bool* is_encoded) -> io::Result<uint64_t> {
   return res;
 }
 
+error_code RdbLoaderBase::HandleCompressedBlob() {
+  if (!zstd_decompress_) {
+    zstd_decompress_.reset(new ZstdDecompressImpl());
+  }
+  // Fetch uncompress blob
+  string res;
+  SET_OR_RETURN(FetchGenericString(), res);
+  // Decompress blob
+  string_view uncompressed_blob;
+  SET_OR_RETURN(zstd_decompress_->Decompress(res), uncompressed_blob);
+
+  // Push uncompressed blob to its membuf
+  uncompressed_mem_buf_.Reserve(uncompressed_blob.size());
+  IoBuf::Bytes dest = uncompressed_mem_buf_.AppendBuffer();
+  CHECK_GE(dest.size(), uncompressed_blob.size());
+  memcpy(dest.data(), uncompressed_blob.data(), uncompressed_blob.size());
```

Comment:
there is some inefficiency here (double write to memory). maybe add todo: to fix interfaces so that we avoid redundant memory copy.

Reply:
I see, I am fixing this now